controllers/capi2argo_reconciler_test.go

```diff
@@ -40,15 +53,27 @@ var _ = BeforeSuite(func() {
 	By("bootstrapping test environment")
 	logf.SetLogger(zap.New(zap.WriteTo(GinkgoWriter), zap.UseDevMode(true)))
 
-	TestEnv = &envtest.Environment{}
+	Scheme = runtime.NewScheme()
+	gv := clusterv1.GroupVersion
+	Scheme.AddKnownTypes(gv, &clusterv1.Cluster{}, &clusterv1.ClusterList{})
+	utilruntime.Must(clientgoscheme.AddToScheme(Scheme))
+	utilruntime.Must(apiextensionsv1.AddToScheme(Scheme))
+	_ = clusterv1alpha4.AddToScheme(Scheme)
```

Comment:
```suggestion
	_ = clusterv1alpha4.AddToScheme(Scheme)
	utilruntime.Must(cluterv1.AddToScheme)
```
Your issue seems to be that you do not have the clutser's cluster-api scheme added.  

This should help get your tests passing.

Reply:
@troy0820 thx for taking a look! I already tried this with no luck. line 58 and above is another approach of registering the correct schemes.